tests/test_samples.py

```diff
@@ -638,3 +638,38 @@ def test_posterior_points():
     ns = NestedSamples(root='./tests/example_data/pc')
     assert_array_equal(ns.posterior_points(), ns.posterior_points())
     assert_array_equal(ns.posterior_points(0.5), ns.posterior_points(0.5))
+
+
+def test_logzero_mask_prior_level():
+    np.random.seed(3)
+    ns0 = NestedSamples(root='./tests/example_data/pc')
+    pi0 = ns0.set_beta(0)
+    NS0 = ns0.ns_output(nsamples=2000)
+    mask = ((ns0.x0 > -0.3) & (ns0.x2 > 0.2) & (ns0.x4 < 3.5)).to_numpy()
+
+    V_prior = pi0[mask].weights.sum() / pi0.weights.sum()
+    V_posterior = ns0[mask].weights.sum() / ns0.weights.sum()
+    logZ_V_mask = NS0.logZ.mean() + np.log(V_posterior) - np.log(V_prior)
+
+    ns1 = merge_nested_samples((ns0[mask],))
+    NS1 = ns1.ns_output(nsamples=2000)
+
+    assert abs(NS1.logZ.mean() - logZ_V_mask) < NS1.logZ.std()
```

Comment:
This tests completely discards the samples excluded by the mask.

This test actually passes, i.e. the `logZ` from prior and posterior volume reweighting lies within 1 sigma of the mean of the recomputed `ns_output()`.

Reply:
This corresponds to the green line in the MWE plots.